### PR TITLE
Fixes sometimes clicking ad notifications doesn't do anything (click not being registered)

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/11042

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/11042

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
CI failed for known speedreader browser test

File: components/brave_ads/browser/ads_service_factory.cc [hunk 1]
```diff
@@ -31,6 +32,10 @@ namespace brave_ads {
 // static
 AdsService* AdsServiceFactory::GetForProfile(
     Profile* profile) {
+  if (profile->IsOffTheRecord() || brave::IsTorProfile(profile)) {
+    return nullptr;
+  }
```
Comment (yrliou) [inline components/brave_ads/browser/ads_service_factory.cc:37]: @tmancey How are the changes in this file related to this PR? This was cleaned up before at https://github.com/brave/brave-core/pull/3597/files#diff-3639f92629488e9f547d8ff7ead4a5f7, and I'm not sure why we need to change it here for this PR's purpose.

Comment (tmancey) [reply components/brave_ads/browser/ads_service_factory.cc:37]: The changes are related as for Tor Profiles the ads service was started and ad notifications were shown. Ad notifications should not be shown for private tabs.

Comment (iefremov) [reply components/brave_ads/browser/ads_service_factory.cc:37]: if `profile->IsOffTheRecord()` doesn't handle Tor profiles, then it looks like we are in trouble, given how many other services could be started for Tor profiles? @yrliou

**Pull request merged**